src/App.js

```diff
@@ -79,7 +80,10 @@ class PageFetchWrapper extends Component {
                 site: this.props.site
             });
         } else {
-            this.setPage(this.props.location.pathname);
+            DotCMSApi.getConfiguration().then(() => {
+                const language = DotCMSApi.page.setLanguage(this.props.location.pathname);
```

Comment:
Is not normal have prefixed `set` function that return something, we have setters and getters for better understanding.